src/app/utils/CompanionApiUtils.js

```diff
@@ -1,9 +1,77 @@
-import React from 'react';
-import { Modifications, Modules, Ships } from 'coriolis-data/dist';
-import Module from '../shipyard/Module';
-import Ship from '../shipyard/Ship';
-import { getBlueprint } from '../utils/BlueprintFunctions';
-import * as ModuleUtils from '../shipyard/ModuleUtils';
+import { Modifications, Modules, Ships } from "coriolis-data/dist";
+import Module from "../shipyard/Module";
+import Ship from "../shipyard/Ship";
+import { getBlueprint } from "../utils/BlueprintFunctions";
+import * as ModuleUtils from "../shipyard/ModuleUtils";
+import { BulkheadNames } from "../shipyard/Constants";
+
+/**
+ * @typedef {keyof SHIP_FD_NAME_TO_CORIOLIS_NAME} CoriolisShipName
+ * @typedef {typeof SHIP_FD_NAME_TO_CORIOLIS_NAME[CoriolisShipName]} EDShipName
+ */
+
+// Mapping from fd's core internal to coriolis and vice versa
+export const CORE_INTERNAL_NAME_MAPPING = {
+  fdToCoriolis: {
+    PowerPlant: "powerPlant",
+    MainEngines: "thrusters",
+    Thrusters: "thrusters",
+    FrameShiftDrive: "frameShiftDrive",
+    LifeSupport: "lifeSupport",
+    Radar: "sensors",
+    FuelTank: "fuelTank",
+    PowerDistributor: "powerDistributor",
+    Armour: "bulkheads",
+    CargoHatch: "cargoHatch",
+  },
+  coriolisToFD: {
+    powerPlant: "PowerPlant",
+    thrusters: "MainEngines",
+    frameShiftDrive: "FrameShiftDrive",
+    lifeSupport: "LifeSupport",
+    sensors: "Radar",
+    fuelTank: "FuelTank",
+    powerDistributor: "PowerDistributor",
+    bulkheads: "Armour",
+    cargoHatch: "CargoHatch",
+    Sensors: "Radar",
+    FSD: "FrameShiftDrive",
+    Thrusters: "MainEngines",
```

Comment:
For some reason I found both of these in the official journals, so I added them here to cover all the bases